### PR TITLE
Improve OpenTelemetry docs, fix #5688

### LINKED ISSUE #5688: OpenTelemetry documentation is misleading with the deprecated Elastic exporter more visible than the OTLP exporter
As shown on https://discuss.elastic.co/t/servermap-module-data-exceptions/278012/7 , the current [Elastic + OpenTelemetry documentation](https://www.elastic.co/guide/en/apm/get-started/current/open-telemetry-elastic.html) is misleading with the deprecated Elastic exporter more visible than the OTLP exporter.

We should clearly highlight that the OpenTelemetry Collector Exporter for Elastic is deprecated and primarily document the OTLP exporter.

We will fix this just after merging https://github.com/elastic/apm-server/pull/5674.

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
* Limit docs on the deprecated OpenTelemetry Collector exporter for Elastic in this guide and focus on the new OTLP solution
* Introduce a "Known limitations" section

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
@EamonnTP @bmorelli25 could you please explain me how we will port this documentation on the 7.14 and later branches?

**Comment by axw** [comment]
@cyrille-leclerc can you please add a known limitation about extrapolating metrics when sampling? i.e. if your OTel traces are sampled < 100%, throughput metrics in the UI will be off. This is because there's no record of the sampling rate in span data; we would need https://github.com/open-telemetry/oteps/pull/148 for this.

**Comment by cyrille-leclerc** [comment]
Thanks @axw , do you mind creating an "Enhancement" request to track the limitation related to metrics accuracy when sampling? maybe in the elastic/apm repo?
Please add the OpenTelemetry label

**Comment by cyrille-leclerc** [comment]
@axw I created it see https://github.com/elastic/apm/issues/472

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:30]:
```suggestion
IMPORTANT: The https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#legacy-opentelemetry-collector-exporter-for-elastic[OpenTelemetry Collector exporter for Elastic]
was deprecated in 7.13 and replaced by the native support of the OpenTelemetry Line Protocol in
Elastic Observability (OTLP). To learn more, see 
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/elasticexporter#migration[migration].
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:76]:
```suggestion
You are now ready to collect traces and <<open-telemetry-elastic-metrics,metrics>> before <<open-telemetry-elastic-verify,verifying metrics>>
and <<open-telemetry-elastic-kibana,visualizing metrics>> in {kib}.
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:81]:
```suggestion
Using the OpenTelemetry collector instances in your architecture, you can connect them to Elastic Observability using the OTLP exporter.
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:117]:
```suggestion
<1> The receivers, such as
the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver[OTLP receiver], that forward data emitted by APM agents or the https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/hostmetricsreceiver[host metrics receiver].
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:118]:
```suggestion
<2> We recommend using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md[Batch processor] and also suggest using the https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiter/README.md[memory limiter processor]. For more information, see https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/README.md#recommended-processors[Recommended processors].
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:119]:
```suggestion
<3> The https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/loggingexporter[logging exporter] is helpful for troubleshooting and supports various logging levels: `debug`, `info`, `warn`, and `error`.
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:120]:
```suggestion
<4> Elastic Observability endpoint configuration. To learn more, see https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter[OpenTelemetry Collector > OTLP gRPC exporter].
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:121]:
```suggestion
<5> Hostname and port of the APM Server endpoint. For example, `elastic-apm-server:8200`.
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:122]:
```suggestion
<6> Credential for Elastic APM {apm-server-ref-v}/secret-token.html[secret token authorization] (`Authorization: "Bearer a_secret_token"`) or {apm-server-ref-v}/api-key.html[API key authorization] (`Authorization: "ApiKey an_api_key"`).
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:123]:
```suggestion
<7> Environment-specific configuration parameters can be conveniently passed in as environment variables documented https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables[here] (e.g. `ELASTIC_APM_SERVER_ENDPOINT` and `ELASTIC_APM_SERVER_TOKEN`).
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:311]:
```suggestion
==== Limitations in 7.13
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:313]: This section title can be removed.

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:315]:
```suggestion
[[elastic-open-telemetry-traces-limitations]]
===== OpenTelemetry traces

```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:317]:
```suggestion
* Traces of applications using `messaging` semantics might be wrongly displayed or not shown in the APM UI. You may only see `spans` coming from such services, but no `transaction` https://github.com/elastic/apm-server/issues/5094[#5094]
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:319]:
```suggestion
* Inability to see Stack traces in spans or, in general, arbitrary span events for applications instrumented with OpenTelemetry https://github.com/elastic/apm-server/issues/4715[#4715]
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:321]:
```suggestion
* Inability in APM views to view the "Time Spent by Span Type" https://github.com/elastic/apm-server/issues/5747[#5747]
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:323]:
```suggestion
* Metrics derived from traces (throughput, latency, and errors) are not accurate when traces are sampled before being ingested by Elastic Observability (ie by an OpenTelemetry Collector or OpenTelemetry APM agent or SDK) https://github.com/elastic/apm/issues/472[#472]

```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:324]:
```suggestion
[[elastic-open-telemetry-metrics-limitations]]
===== OpenTelemetry metrics

```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:327]:
```suggestion
* OpenTelemetry histogram metrics, https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/DoubleValueRecorder.html[`DoubleValueRecorder`]
and https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-api/latest/io/opentelemetry/api/metrics/LongValueObserver.html[`LongValueRecorder`], are not yet supported https://github.com/elastic/apm-server/issues/3195[#3195]
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:329]:
```suggestion
* Inability to see JVM metrics in Elastic APM UI for Java applications instrumented with OpenTelemetry https://github.com/elastic/apm-server/issues/4919[#4919]
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:331]:
```suggestion
* Inability to see host metrics in Elastic Metrics Infrastructure view when using the OpenTelemetry Collector host metrics receiver https://github.com/elastic/apm-server/issues/5310[#5310]

```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:332]:
```suggestion
[[elastic-open-telemetry-logs-limitations]]
===== OpenTelemetry logs

```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (EamonnTP) [inline docs/guide/opentelemetry-elastic.asciidoc:334]:
```suggestion
* OpenTelemetry logs are not yet supported https://github.com/elastic/apm-server/issues/5491[#5491]
```

**Review by EamonnTP — COMMENTED** [review]
Looks great, just some minor suggestions. Note that "Positive Rate" will need updated to "Counter Rate" in the Visualize in Kibana section.

**Comment by EamonnTP** [comment]
> @EamonnTP @bmorelli25 could you please explain me how we will port this documentation on the 7.14 and later branches?

@cyrille-leclerc We would use the back port tool for back porting PRs. This is something Brandon and I could help out with. After merging, I take it this PR needs back ported to the `master`, `7.x`, and `7.14` branches?

**Comment by EamonnTP** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (bmorelli25) [reply docs/guide/opentelemetry-elastic.asciidoc:313]:
```suggestion
```

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:36]: Great diagram

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (bmorelli25) [inline docs/guide/opentelemetry-elastic.asciidoc:83]:
```suggestion
```

**Review by bmorelli25 — APPROVED** [review]
I committed the rest of @EamonnTP's suggestions as it looks like you just missed the expand comment button. Otherwise, this looks good! Thanks, Cyrille.

**Comment by bmorelli25** [comment]
I'll merge and cherry-pick when the tests are green.

File: docs/guide/opentelemetry-elastic.asciidoc
Comment (cyrille-leclerc) [reply docs/guide/opentelemetry-elastic.asciidoc:36]: thanks

**Comment by cyrille-leclerc** [comment]
Thanks Brandon

**Pull request merged**
